contracts/profiles/AddrResolver.sol

```diff
@@ -26,10 +28,26 @@ contract AddrResolver is ResolverBase {
      * @return The associated address.
      */
     function addr(bytes32 node) public view returns (address) {
-        return addresses[node];
+        bytes memory a = addr(node, COIN_TYPE_ETH);
+        if(a.length == 0) {
+            return address(0);
+        }
+        return bytesToAddress(a);
+    }
+
+    function setAddr(bytes32 node, uint coinType, bytes memory a) public authorised(node) {
```

Comment:
Would it make sense to have some form of Code Repository with codes to strings?

Reply:
Definitely - but offchain.